src/git_repositories.ml

```diff
@@ -76,6 +76,24 @@ module Repositories = struct
     Current.Process.check_output ~cwd ~cancellable:true ~job ("", [|"git"; "rev-parse"; "HEAD"|]) >>!= fun hash ->
     Lwt.return (Ok (String.trim hash))
 
+  let get_latest_release_hash ~job ~repo =
+    Current.Process.with_tmpdir ~prefix:"git-checkout" @@ fun cwd ->
+    Current.Process.exec ~cwd ~cancellable:true ~job ("", [|"git"; "clone"; repo; "."|]) >>!= fun () ->
+    Current.Process.check_output ~cwd ~cancellable:true ~job ("", [|"git"; "tag"; "--format"; "%(objectname) %(refname:strip=2)"|]) >>!= fun all_tags ->
+    let rec parse = function
+      | "" -> []
+      | s -> Scanf.sscanf s "%s %s %s@!" (fun sha tag r -> (tag, sha) :: parse r) in
+    let hash = parse all_tags
+      |> List.filter_map (fun (tag, sha) ->
+        match tag.[0] with
+        | '0'..'9' ->
+          let normalised = String.split_on_char '-' tag |> String.concat "~" in
+          Some (OpamVersion.of_string normalised, sha)
+        | _ -> None)
+      |> List.sort (fun (v1, _) (v2, _) -> OpamVersion.compare v2 v1)
+      |> List.hd |> snd in
+    Lwt.return (Ok hash)
```

Comment:
I think this probably fine as is, but it may be worth making this more robust, and returning an `Error` with a useful message in case we get an empty list of results, otherwise, if something goes wrong in invoking the command we'll have to puzzle out an `Exception: Failure "hd"` from line 94.